sklearn/preprocessing/imputation.py

```diff
@@ -357,6 +379,17 @@ def transform(self, X):
 
             X.data[mask] = astype(valid_statistics[indexes], X.dtype,
                                   copy=False)
+
+            if self.add_missing_indicator:
+                mask_matrix = X.__class__((mask, X.indices.copy(), X.indptr.copy()), shape=X.shape, dtype=X.dtype)
+                mask_matrix.eliminate_zeros()  # removes explicit False entries to make the matrix sparser
+                features_with_missing_values = np.where(np.logical_not(sparse_min_max(mask_matrix, 0)[1]))
```

Comment:
No: sparse matrices build on a (now outdated) concept of a matrix type in numpy. These matrices are always 2d, and reduction operations on them similarly produce a matrix (or rather a 2d column- or row-vector). So reducing a sparse matrix over axis=0 will produce a 2d row vector.